src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/ReferencesPage/ImplicitUsingsValueProvider.cs

```diff
@@ -0,0 +1,200 @@
+
+                    if (existingUsingsOfIncludeToRemove.Count > 0)
+                    {
+                        project.RemoveItems(
+                            project.GetItems("Using")
+                                .Where(i => string.Equals(usingToSet.Include, i.EvaluatedInclude, StringComparisons.ItemNames) && !i.IsImported)
+                        );
+                    }
+                    
+                    if (usingToSet.Include.Length > 0 && (usingToSet.Alias is null || usingToSet.Alias.Length > 0))
+                    {
+                        var usingMetadata = new List<KeyValuePair<string, string>>();
+                        if (usingToSet.Alias is { } alias)
+                        {
+                            usingMetadata.Add(new KeyValuePair<string, string>("Alias", alias));
+                        }
+
+                        if (usingToSet.IsStatic)
+                        {
+                            usingMetadata.Add(new KeyValuePair<string, string>("Static", usingToSet.IsStatic.ToString()));
+                        }
+                        
+                        project.AddItem("Using", usingToSet.Include, usingMetadata);
+                    }
+                }
+            });
+        }
+
+        return null;
+    }
+
+    internal class ImplicitUsing
+    {
+        public string Include { get; }
+        public string? Alias { get; }
+        public bool IsStatic { get; }
+        public bool IsReadOnly { get; }
+        
+        public ImplicitUsing(string include, string? alias, bool isStatic, bool isReadOnly)
+        {
+            Include = include;
+            Alias = alias;
+            IsStatic = isStatic;
+            IsReadOnly = isReadOnly;
+        }
+
+        protected bool Equals(ImplicitUsing other)
+        {
+            return Include == other.Include && Alias == other.Alias && IsStatic == other.IsStatic && IsReadOnly == other.IsReadOnly;
+        }
+
+        public override bool Equals(object? obj)
+        {
+            if (ReferenceEquals(null, obj))
+            {
+                return false;
+            }
+
+            if (ReferenceEquals(this, obj))
+            {
+                return true;
+            }
+
+            if (obj.GetType() != GetType())
+            {
+                return false;
+            }
+            
+            return Equals((ImplicitUsing)obj);
+        }
+
+        public override int GetHashCode()
+        {
+            unchecked
+            {
+                int hashCode = Include.GetHashCode();
+                hashCode = (hashCode * 397) ^ (Alias != null ? Alias.GetHashCode() : 0);
+                hashCode = (hashCode * 397) ^ IsStatic.GetHashCode();
+                hashCode = (hashCode * 397) ^ IsReadOnly.GetHashCode();
+                return hashCode;
```

Comment:
We may have a utility type that can help with this.

Reply:
That is, you may be able to replace all this with a method call.

Reply:
It doesn't look like there is :/